### ISSUE TITLE
Find longterm frontend solution 

### ISSUE THREAD

**Issue by leepeuker** [leepeuker/movary#225]
The current frontend consists of hacky twig templates and a few thrown together js/css files, which are not so great to maintain.  

Possible solutions for this I see are:
- refactor and improve the twig templates and use something like webpack to manage the js code
- use a framework like vue.js or next.js for the frontend

Right now I would favor building a vue.js proof of concept, mainly because I had the most experience with it in the past. Not sure when I will have the time to look into that more, help appreciated.

**Comment by perebusquets** [comment]
Hi! I've just found this media library and I would be willing to contribute (specially on this topic). 

Mainly I have experience with React but I would also be willing to get hands on with VueJS if need be. Just to make sure we get the right framework/setup for this, I will ask some questions if you don't mind:

1. In the post, you propose NextJS instead of React as an alternative, which is often used for server side rendering. Is this because of some need/functionality? Or just because of the batteries included nature of NextJS?
2. Would you be willing to accept a poc made with React? (in this case I could provide everything set up, including translations, build tools, authentication management, etc pretty quick, as it's what I'm most familiar with).

IMHO, regardless of what framework we use, I think the best path would be a SPA, which gets bundled into a single JS file. This way we could send it in a twig template, and there wouldn't be any need to run an extra docker container or having node installed. It also brings the possibility of easily implementing it as a PWA, having a good mid term solution between only being web based and having a mobile app.

Once a framework is decided, I would also like to suggest that we use a component library for speed of development. With this one, I'm also open to suggestions!

**Comment by leepeuker** [comment]
To be honest I cannot answer most of the question because I do not know the answer without more recherche.

> IMHO, regardless of what framework we use, I think the best path would be a SPA, which gets bundled into a single JS file. This way we could send it in a twig template, and there wouldn't be any need to run an extra docker container or having node installed. It also brings the possibility of easily implementing it as a PWA, having a good mid term solution between only being web based and having a mobile app.

That sounds like a good approach to me, any POC in that direction would be very welcome

**Comment by perebusquets** [comment]
Here's the promised proof of concept! If there are any questions, just ask away!

**Comment by leepeuker** [comment]
Thank you @perebusquets for the POC, that helped me a lot to get a better feeling for the topic. 

I do not think that we will move forward with it right now, I want to focus on making the current setup more stable for version 1.0, but I will keep this open for later versions were we can focus on modernizing the frontend.

**Comment by perebusquets** [comment]
Makes sense, if any questions arise please don't hesitate to contact me!

**Comment by JVT038** [comment]
@leepeuker Is there any update on this? I think it would be the best to do this ASAP, because of the ever-growing codebase. The longer we wait, the more difficult it will become to overhaul / rewrite the frontend (and probably a part of the backend).
